### ISSUE TITLE
Incomplete Download

### ISSUE THREAD

**Issue by pdl86** [opf/OpenProjectExcel#27]
Hello 

I have a project ( scrum) , with 129 items,

the Excel macro is able to correct and corretcly retrieve all field headrers, . 
There is one Custom field.
Once setting up the additional field) , the dowload request return corret ly the 129 record, ofr ID and Subjetc , but only some of the data for Type ( a standrd field) and Order ( a custom field) .

this make the tool difficult to use.

Note taht the view use has hierarcgy display in use and is sorted on the "Ordre" field.

![image](https://user-images.githubusercontent.com/30148795/101143490-3f5ae000-3617-11eb-98bf-d4a2ae30e3bf.png)

thanks in advance for looking at the issue
Philippe

**Comment by bpetraus** [comment]
Hi @pdl86 ,

thats suprising. Can you please try the following:

1 clear all rows and download the workpackages (save the result, like in another excel file)
2 go to openproject and change for a not working workpackage (e.g. "175 KPI Sales admin V" Operationell")  "Ordre" and "Type" to a value. that works for other workackages (like "A1.4" for "Ordre" and "Feature" for "Type"
3 clear all rows again and download the workpackages again
4 tell me the result

Best

Bernd

**Comment by pdl86** [comment]
Hi Bernd

I will try but on 0.5.22  ( as I have a problem 0.5.23 ( see #29) 

regards
Philippe

**Comment by pdl86** [comment]
Hi Bernd

I have opne the Excle file, set up the macro ( as default, then Accept) ,
git the headers
select "Ordre" and "Type"
request a download : incomplete

empty all lines 
save the Xcel under a new name

recativate the macro ( Ctrl+b) ,  display the choosen projetc : OK parameters set-up
request a download 

and then magic it works !
( all f value in all field) 

this WITHOUT any change on the OpenProjetc side.

It seems that saving the Xcel file with the default value, clean something

At leats I have a work around ( thank for indicating a good directtion).

So in summary, I will recommend to use  intitialise the macro,  get the headers, and the save it as a new sheet, and then download value in the new sheet.

Your thoughts ? 

regards
Philippe

**Comment by bpetraus** [comment]
Hi @pdl86 ,

right at your first step (before you first requested a downalod), did you opened a fresh file or where there already workpackages?

Best

Bernd

**Comment by pdl86** [comment]
Hi 
now the it was a fresh file.

after few more test ( with now 0.5.24) , it seems that if you download a first time it is "wrong", then your clear lines et re-query a second time and then it is OK .

but then if you request , new lines are added after existing ones, no cleaning . 

So it seems it has something to do with clearing existing lines before filling them again with the query results.

If I clear all lines and re-issue a query , it is fine each time.

regards


regards
philippe

**Comment by bpetraus** [comment]
Hi @pdl86 ,

"but then if you request , new lines are added after existing ones, no cleaning ."

Thats intended. In any case, the excel file will not update existing lines. It just downloads another set of workpackages, starting at the end of the list (even when the workpackages are the same). So its always a good idea to clear the list before re-downloading.

"after few more tests ( with now 0.5.24) , it seems that if you download a first time it is "wrong","

That is super strange. I am not able to reproduce this behaviour and I unfortunately dont have any clue here. :(  Could you maybe provide a short video or something?

Best 

Bernd

**Comment by pdl86** [comment]
Hi Bernd

It seems it ocurrs ONLY the very first time, and ONLY if you did not save the file. ( so the original xlsm file)
 Afterward after a "line cleaning" it works all time.
( or if you save the Xcel file, close it and reopen the newly namle file , it works )

regards
Philippe

NOT I use Excel 10, not 16
